google/cloud/storage/examples/storage_async_samples.cc

```diff
@@ -200,17 +202,116 @@ void ReadObjectWithOptions(
   std::cout << "The object contains " << count << " lines\n";
 }
 
+void WriteObject(google::cloud::storage_experimental::AsyncClient& client,
+                 std::vector<std::string> const& argv) {
+  //! [write-object]
+  namespace gcs = google::cloud::storage;
+  namespace gcs_ex = google::cloud::storage_experimental;
+  auto coro = [](gcs_ex::AsyncClient& client, std::string bucket_name,
+                 std::string object_name, std::string const& filename)
+      -> google::cloud::future<gcs::ObjectMetadata> {
+    std::ifstream is(filename);
+    if (is.bad()) throw std::runtime_error("Cannot read " + filename);
+    auto [writer, token] = (co_await client.WriteObject(std::move(bucket_name),
+                                                        std::move(object_name)))
+                               .value();
+    while (token.valid()) {
+      std::vector<char> buffer(1024 * 1024);
+      is.read(buffer.data(), buffer.size());
+      buffer.resize(is.gcount());
+      token = (co_await writer.Write(std::move(token),
+                                     gcs_ex::WritePayload(std::move(buffer))))
+                  .value();
+      if (is.eof()) break;
```

Comment:
same comment about making this a loop condition.

Reply:
Fixed.